### PR TITLE
feat: support multiple graphhopper instances based on bbox

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR refactors valhalla to support multiple GraphHopper instances based on bounding box. All bounding boxes and servers are configurable via the environment config file.

### CODE REVIEW

File: lib/editor/util/map.js [hunk 1]
```diff
@@ -44,11 +44,14 @@ type R5Response = {
   }>
 }
 
+export const coordIsOutOfBounds = (coords: LatLng, bounds: any) => {
```
Comment (philip-cline) [inline lib/editor/util/map.js]: Are we expecting a Leaflet type here for `bounds` rather than `any`?

**Review by philip-cline — APPROVED** [review]
Nice, great to see this moving forward.

File: lib/editor/util/map.js [hunk 2]
```diff
@@ -44,11 +44,14 @@ type R5Response = {
...
+    coords.lat < bounds.getSouth() ||
+    bounds.lng > bounds.getEast() ||
+    bounds.lng < bounds.getWest()
+}
 export const stopIsOutOfBounds = (stop: GtfsStop, bounds: any) => {
-  return stop.stop_lat > bounds.getNorth() ||
-    stop.stop_lat < bounds.getSouth() ||
-    stop.stop_lon > bounds.getEast() ||
-    stop.stop_lon < bounds.getWest()
+  return coordIsOutOfBounds({lat: stop.stop_lat, lng: stop.stop_lon})
```
Comment (philip-cline) [inline lib/editor/util/map.js]: Do we need to pass `bounds` to `coordIsOutOfBounds`? I'm getting a typeError when running this.

**Review by philip-cline — CHANGES_REQUESTED** [review]
Just one comment to address actually

File: lib/editor/util/map.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/editor/util/map.js]: I copied the type from the `stopIsOutOfBounds` this is adapted from. However `bounds` passes flow check so I've made the change

File: lib/editor/util/map.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply lib/editor/util/map.js]: Yep slept right through that one thanks for noticing! I guess the thing I did to test that method was working wasn't working.

File: lib/editor/util/map.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/editor/util/map.js]: Fixed in 2bae81f17a9497173531d0c7261e7cdc1f2982a2

File: lib/editor/util/map.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply lib/editor/util/map.js]: Fixed in 2bae81f17a9497173531d0c7261e7cdc1f2982a2

**Review by philip-cline — APPROVED** [review]
Working well for me now!

File: lib/scenario-editor/utils/valhalla.js [hunk 3]
```diff
@@ -5,11 +5,13 @@ import {decode as decodePolyline} from 'polyline'
 import {isEqual as coordinatesAreEqual} from '@conveyal/lonlat'
 import qs from 'qs'
 import lineString from 'turf-linestring'
+import L from 'leaflet'
```
Comment (binh-dam-ibigroup) [inline lib/scenario-editor/utils/valhalla.js]: Sort imports (by package)

File: lib/scenario-editor/utils/valhalla.js [hunk 4]
```diff
@@ -53,6 +55,12 @@ type GraphHopperResponse = {
   paths: Array<Path>
 }
 
+type GraphHopperAlternativeServer = {
```
Comment (binh-dam-ibigroup) [inline lib/scenario-editor/utils/valhalla.js]:
```suggestion
type GraphHopperAlternateServer = {
```

**Review by binh-dam-ibigroup — APPROVED** [review]
Looks fine and doesn't seem to interfere with anything else.

File: lib/scenario-editor/utils/valhalla.js (hunk 3 above)
Comment (miles-grant-ibigroup) [reply lib/scenario-editor/utils/valhalla.js]: Sorted in e6c88ec1

File: lib/scenario-editor/utils/valhalla.js (hunk 4 above)
Comment (miles-grant-ibigroup) [reply lib/scenario-editor/utils/valhalla.js]: Good call! Renamed in e6c88ec1

**Pull request merged**